tests/configuration_test.py

```diff
@@ -15,53 +15,46 @@ async def test_config(servo_yaml: pathlib.Path) -> None:
     )
     config = config_model.generate()
     servo_yaml.write_text(config.yaml())
-
-    optimizer = servo.configuration.Optimizer(
-        id="servox.opsani.com/tests",
-        token="00000000-0000-0000-0000-000000000000",
-    )
-    assembly_ = await servo.assembly.Assembly.assemble(
-        config_file=servo_yaml, optimizer=optimizer
-    )
+    assembly_ = await servo.assembly.Assembly.assemble(config_file=servo_yaml)
 
 
 @pytest.mark.parametrize(
     ("name"), ["a", "1234", "app-name", "APP-NAME", "name", "this.that.the.other"]
 )
 def test_validate_name(name: str):
     # will raise on failure
-    servo.configuration.Optimizer(id=f"test.com/{name}", token="foo")
+    servo.configuration.OpsaniOptimizer(id=f"test.com/{name}", token="foo")
```

Comment:
It seems this test was created as a validation of the name regex constraint applied to the OpsaniOptimizer. Said constraint never really made much sense to me considering the optimizer ID (org/name) are provisioned by the backend instead of chosen by the end user via servo. Such validations should be (and are) enforced during optimizer creation and would only cause (and have caused) needless issues with servo configuration. Hence I have applied no such constraints to the new AppdynamicsOptimizer configuration and this test is not really applicable to it.

https://github.com/opsani/servox/blob/666222904ea163d3727e62c417f0d4852278d1d3/servo/configuration.py#L50